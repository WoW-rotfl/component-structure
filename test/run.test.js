const { promisify } = require('util');
const mkdirp = require('mkdirp');
const exec = promisify(require('child_process').exec);
const fs = require('fs');
const path = require('path');
const rimraf = require('rimraf');

const componentsPath = path.resolve('./test/components');

beforeAll(() => {
  const LinkPath = path.join(componentsPath, 'Link');
  mkdirp.sync(LinkPath);
  fs.writeFile(path.join(LinkPath, 'index.js'));
});

afterAll(() => {
  rimraf.sync(componentsPath);
});

it('shows an error on define structure', async () => {
  await exec('./index.js -p test').catch(result => {
    expect(result.stderr).toMatchSnapshot();
  });

  await exec('./index.js -p test --structure').catch(result => {
    expect(result.stderr).toMatchSnapshot();
  });
});

it('shows an error on define destination path', async () => {
  await exec('./index.js -s test').catch(result => {
    expect(result.stderr).toMatchSnapshot();
  });

  await exec('./index.js -s test --path').catch(result => {
    expect(result.stderr).toMatchSnapshot(); 
  });
});

it('should create Button component', async () => {
  const { stdout } = await exec('./index.js Button -p ./test/components -s index.js .js __tests__/.test.js');

  expect(stdout).toMatchSnapshot();
  expect(fs.existsSync(path.join(componentsPath, 'Button'))).toBeTruthy();

});

it('should fail create Link component', async () => {
  const { stdout } = await exec('./index.js Link -p ./test/components -s index.js');

  expect(stdout).toMatchSnapshot();
});
