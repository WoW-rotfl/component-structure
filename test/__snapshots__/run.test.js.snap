// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create Button component 1`] = `
"

SUCCESS>>> Component Button created in ./test/components


"
`;

exports[`should fail create Link component 1`] = `
"

FAIL>>>>>> Component Link already exists in ./test/components


"
`;

exports[`shows an error on define destination path 1`] = `
"Options:
  --help           Show help                                           [boolean]
  --version        Show version number                                 [boolean]
  --structure, -s  Define component structure e.g.
                   .js .stories.js __tests__/.test.js ->
                   component-name/component-name.js,
                   component-name/component-name.stories.js,
                   component-name/__tests__/component-name.test.js
                                                              [array] [required]
  --path, -p       Set component destination path            [string] [required]

Missing required argument: path
Provide component destination relative path
"
`;

exports[`shows an error on define destination path 2`] = `
"Options:
  --help           Show help                                           [boolean]
  --version        Show version number                                 [boolean]
  --structure, -s  Define component structure e.g.
                   .js .stories.js __tests__/.test.js ->
                   component-name/component-name.js,
                   component-name/component-name.stories.js,
                   component-name/__tests__/component-name.test.js
                                                              [array] [required]
  --path, -p       Set component destination path            [string] [required]

Not enough arguments following: path
"
`;

exports[`shows an error on define structure 1`] = `
"Options:
  --help           Show help                                           [boolean]
  --version        Show version number                                 [boolean]
  --structure, -s  Define component structure e.g.
                   .js .stories.js __tests__/.test.js ->
                   component-name/component-name.js,
                   component-name/component-name.stories.js,
                   component-name/__tests__/component-name.test.js
                                                              [array] [required]
  --path, -p       Set component destination path            [string] [required]

Missing required argument: structure
Provide component structure
"
`;

exports[`shows an error on define structure 2`] = `
"Options:
  --help           Show help                                           [boolean]
  --version        Show version number                                 [boolean]
  --structure, -s  Define component structure e.g.
                   .js .stories.js __tests__/.test.js ->
                   component-name/component-name.js,
                   component-name/component-name.stories.js,
                   component-name/__tests__/component-name.test.js
                                                              [array] [required]
  --path, -p       Set component destination path            [string] [required]
"
`;
